src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/LinkOutputImpl.java

```diff
@@ -51,8 +51,7 @@ public LinkOutputImpl() {
 
     @Override
     public void append(Object o) {
-        output.append(o instanceof String ?
-            (String) o : ((LinkOutputImpl)o).toString());
+        output.append(o.toString());
```

Comment:
There may be a change in behavior here, for "invalid" arguments, which will now use `.toString()` instead of throwing CCE.

Reply:
The documentation header is a standard langtools warnings to all those people that rely on access to javac and javadoc internals.

Class not used ... wow, correct. I'll file an RFE to investigate and remove it.

Reply:
> There may be a change in behavior here, for "invalid" arguments, which will now use `.toString()` instead of throwing CCE.

Jon is right about the behavior change. That was a sloppy suggestion on my part. Sorry, Ian; you should revert that to your original version. Jon has created an RFR to remove LinkOutputImpl and LinkOutput: https://git.openjdk.java.net/jdk/pull/4121

Reply:
Correction. Instead of reverting to your original version, you should change it to this:

```
    @Override
    public void append(Object o) {
        output.append(o instanceof String s ? s : ((LinkOutputImpl) o).toString());
    }
```